src/System.Runtime/tests/System/GCTests.netcoreapp.cs

```diff
@@ -23,5 +23,17 @@ public static void GetAllocatedBytesForCurrentThread(int size)
             Assert.True((end - start) > size, $"Allocated too little: start: {start} end: {end} size: {size}");
             Assert.True((end - start) < 5 * size, $"Allocated too much: start: {start} end: {end} size: {size}");
         }
+
+        [Fact]
+        public static void GetGCMemoryInfo()
+        {
+            GCMemoryInfo memoryInfo = GC.GetGCMemoryInfo();
+
+            Assert.True(memoryInfo.HighMemoryLoadThresholdBytes > 0);
```

Comment:
Are we going to have more thorough tests for this as well?

Reply:
What would you have in mind? This is tricky in nature to test since we don't want to rely on specific machine resources, and it's hard to predict specific values based on GC and other external factors.

Reply:
Some ideas:
- Allocate and keep alive a lot and see that the heap size goes up
- Introduce a lot of GC heap fragmentation (e.g. using pinning) and observe that the fragmentation goes up.

Reply:
See https://github.com/dotnet/corefx/pull/37092/files#diff-0156a0e8dfd6a402846dd4059bb16efaR41 and https://github.com/dotnet/corefx/pull/37092/files#diff-0156a0e8dfd6a402846dd4059bb16efaR77. Please let me know if these tests cover what you had in mind.